core/src/main/java/com/linecorp/armeria/internal/metric/RequestMetricSupport.java

```diff
@@ -33,63 +33,50 @@
 import io.micrometer.core.instrument.DistributionSummary;
 import io.micrometer.core.instrument.MeterRegistry;
 import io.micrometer.core.instrument.Timer;
-import io.netty.util.AttributeKey;
 
 /**
  * Collects the metric data and stores it into the {@link MeterRegistry}.
  */
 public final class RequestMetricSupport {
 
-    private static final AttributeKey<RequestMetrics> ATTR_REQUEST_METRICS =
-            AttributeKey.valueOf(RequestMetricSupport.class, "REQUEST_METRICS");
-
-    private static final RequestMetrics PLACEHOLDER = new RequestMetricsPlaceholder();
-
     public static void setup(RequestContext ctx, MeterIdPrefixFunction meterIdPrefixFunction) {
-        if (ctx.hasAttr(ATTR_REQUEST_METRICS)) {
-            return;
-        }
-
-        // Set the attribute to the placeholder so that calling this method again has no effect.
-        // The attribute will be set to the real one later in onRequestStart()
-        ctx.attr(ATTR_REQUEST_METRICS).set(PLACEHOLDER);
```

Comment:
We need at least a boolean attribute so that calling `setup()` multiple times does not add the log listener multiple times:

```java
private static final AttributeKey<Boolean> ATTR_REQUEST_METRICS_SET = ...;

public static void setup(...) {
    if (ctx.hasAttr(ATTR_REQUEST_METRICS_SET) { return; }
    ctx.attr(ATTR_REQUEST_METRICS_SET).set(true);
    ...
}
```